File: Java/BinarySearch.java
public class BinarySearch {
    public static void main(String args[]) {
        int arr[] = { 10, 20, 30, 40, 50 };
        int key = 30;
        int last = arr.length - 1;
        bSearch(arr, 0, last, key);
    }


    static void bSearch(int arr[], int first, int last, int key) {
        int mid = (first + last) / 2;
        while (first <= last) {
            if (arr[mid] < key) {
                first = mid + 1;
            } else if (arr[mid] == key) {
                System.out.println("Element is found at index: " + mid);
                break;
            } else {
                last = mid - 1;
            }
            mid = (first + last) / 2;
        }
        if (first > last) {
            System.out.println("Element is not found!");
        }
    }
   
}

File: Hacktoberfest2023-main/Add Code Here/Javascript/Binary Search.js
function binary_search(x,f)
{
    var l = 0;
    var h = x.length-1;
    var flag = false;

    while(l<h)
    {
        let mid = Math.floor(((l+h) / 2));
        if(x[mid] == f)
        {
            flag = true;
            break;
        }else
        { 
            if(x[mid] < f)
            {
                l = mid;
            }else if(x[mid] > f)
            {
                h = mid;
            }
            
        }
    }
    
    if(flag)
    {
        console.log(f + " has been found!");
    }else
    {
        console.log(f + " is not found!");
    }
}

const arr =  [1,3,5,6,7,77,88,99];
var find = 1;
binary_search(arr, find);